### ISSUE TITLE
Error message shows up in a fresh env

### ISSUE THREAD

**Issue by evelinec** [OpenLiberty/guide-rest-client-angular#41]
Start the app server on this guide and visit the `http://localhost:9080/artists` URL for the very first time in a fresh env and the follow error shows up. Found on Mac, Linux and Windows: 

```
[INFO] [WARNING ] SRVE0190E: File not found: /app
[INFO] [ERROR   ] Error Page Exception: 
[INFO]                                                                                                                guide-rest-client-angular
[INFO]                                                                                                                /
[INFO]                                                                                                                Error Page Exception
[INFO]                                                                                                                com.ibm.ws.webcontainer.webapp.WebAppErrorReport: java.io.FileNotFoundException: SRVE0190E: File not found: /app
[INFO] 	at com.ibm.ws.webcontainer.extension.DefaultExtensionProcessor.handleRequest(DefaultExtensionProcessor.java:493)
[INFO] 	at [internal classes]
[INFO] Caused by: java.io.FileNotFoundException: SRVE0190E: File not found: /app
[INFO] 	... 3 more
[INFO] 
```

```
[2019-11-28, 12:02:22:365 EST] 00000049 com.ibm.ws.webcontainer.extension                            W SRVE0190E: File not found: /app
[2019-11-28, 12:02:22:613 EST] 00000049 com.ibm.ws.logging.internal.impl.IncidentImpl                I FFDC1015I: An FFDC Incident has been created: "java.io.FileNotFoundException: SRVE0190E: File not found: /app com.ibm.ws.webcontainer.webapp.WebApp.handleError 912" at ffdc_19.11.28_12.02.22.0.log
[2019-11-28, 12:02:22:618 EST] 00000049 com.ibm.ws.webcontainer.webapp                               E Error Page Exception: 
                                                                                                               guide-rest-client-angular
                                                                                                               /
                                                                                                               Error Page Exception
                                                                                                               com.ibm.ws.webcontainer.webapp.WebAppErrorReport: java.io.FileNotFoundException: SRVE0190E: File not found: /app
	at com.ibm.ws.webcontainer.webapp.WebApp.sendError(WebApp.java:4355)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleException(WebApp.java:5106)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:5086)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:1007)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1136)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:417)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:376)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:547)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:481)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:346)
	at com.ibm.ws.http.channel.internal.inbound.HttpICLReadCallback.complete(HttpICLReadCallback.java:70)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:503)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:573)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:954)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1043)
	at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:239)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.FileNotFoundException: SRVE0190E: File not found: /app
	at com.ibm.ws.webcontainer.extension.DefaultExtensionProcessor.handleRequest(DefaultExtensionProcessor.java:493)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1218)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1002)
	at com.ibm.ws.webcontainer.webapp.WebAppRequestDispatcher.dispatch(WebAppRequestDispatcher.java:1410)
	at com.ibm.ws.webcontainer.webapp.WebAppRequestDispatcher.forward(WebAppRequestDispatcher.java:171)
	at com.ibm.ws.webcontainer.webapp.WebApp.sendError(WebApp.java:4310)
	... 20 more

[2019-11-28, 12:03:30:072 EST] 0000001d com.ibm.ws.kernel.launch.internal.ServerCommandListener      A CWWKE0055I: Server shutdown requested on Thursday, November 28, 2019 at 12:03 p.m.. The server defaultServer is shutting down.
```

Afterwards, it's not reproducible.

**Comment by realModusOperandi** [comment]
I'm having trouble reproducing this. Can you give me a step-by-step of commands you run to encounter this? Given the transient nature, it's almost as if the app isn't fully loaded (or built?) when you try to access the URL but some asynchronous process finishes up soon after....

**Comment by evelinec** [comment]
This is a tough one, as I can't reproduced after I ran the guide once on an environment. Below are the test steps we did, we could see this error in all environments (Mac, Linux and Windows), only the first time we ran this guide. 

1. Run `git clone https://github.com/openliberty/guide-rest-client-angular.git --branch dev-mode` command from a command prompt
2. Go to `cd guide-rest-client-angular/finish` directory
3. Start the app server by starting the dev mode with this command: `mvn liberty:dev`
4. Visit `http://localhost:9080/artists` URL, by clicking on this link to open on a web browser. 

On the command prompt where you have the dev mode running, you'll see the reported error messages.

**Comment by realModusOperandi** [comment]
Ok, I followed those steps and I get this output from Maven (added at end of post)

What I'm not seeing is any invocation of the maven-frontend-plugin to build the frontend and copy it to the right place. When I access the /artists endpoint, I don't get any errors, but if I try to access /app I see the errors you mention:

```
[INFO] [WARNING ] SRVE0190E: File not found: /app
[INFO] [WARNING ] SRVE0190E: File not found: /app
[INFO] [ERROR   ] Error Page Exception:
[INFO]                                                                                                                guide-rest-client-angular
[INFO]                                                                                                                /
[INFO]                                                                                                                Error Page Exception
[INFO]                                                                                                                com.ibm.ws.webcontainer.webapp.WebAppErrorReport: java.io.FileNotFoundException: SRVE0190E: File not found: /app
[INFO] 	at com.ibm.ws.webcontainer.extension.DefaultExtensionProcessor.handleRequest(DefaultExtensionProcessor.java:493)
[INFO] 	at [internal classes]
[INFO] Caused by: java.io.FileNotFoundException: SRVE0190E: File not found: /app
[INFO] 	... 3 more
[INFO]
```

I think the process-resources phase isn't invoked automatically by the Liberty plugin. Invoking it by issuing package or process-resources before the dev mode avoids this issue. Not sure whether that's a problem with the liberty plugin or just a necessary step to getting everything compiled.

Full output of initial `mvn liberty:dev`:
```
(14:57:37) ~/Projects/temp/test1/finish
[liam@lwestby-ws] -> mvn liberty:dev
[INFO] Scanning for projects...
[INFO]
[INFO] ----------< io.openliberty.guides:guide-rest-client-angular >-----------
[INFO] Building guide-rest-client-angular 1.0-SNAPSHOT
[INFO] --------------------------------[ war ]---------------------------------
[INFO]
[INFO] --- liberty-maven-plugin:3.1:dev (default-cli) @ guide-rest-client-angular ---
[INFO] CWWKM2102I: Using artifact based assembly archive : io.openliberty:openliberty-kernel:null:19.0.0.11:zip.
[INFO] CWWKM2102I: Using installDirectory : /Users/liam/Projects/temp/test1/finish/target/liberty/wlp.
[INFO] CWWKM2102I: Using serverName : defaultServer.
[INFO] CWWKM2102I: Using serverDirectory : /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer.
[INFO] Running maven-compiler-plugin:compile
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 3 source files to /Users/liam/Projects/temp/test1/finish/target/classes
[INFO] Running maven-resources-plugin:resources
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/liam/Projects/temp/test1/finish/src/main/resources
[INFO] Running maven-compiler-plugin:testCompile
[INFO] No sources to compile
[INFO] Running maven-resources-plugin:testResources
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/liam/Projects/temp/test1/finish/src/test/resources
[INFO] Running liberty:create
[INFO] CWWKM2102I: Using artifact based assembly archive : io.openliberty:openliberty-kernel:null:19.0.0.11:zip.
[INFO] CWWKM2102I: Using installDirectory : /Users/liam/Projects/temp/test1/finish/target/liberty/wlp.
[INFO] CWWKM2102I: Using serverName : defaultServer.
[INFO] CWWKM2102I: Using serverDirectory : /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer.
[INFO] CWWKM2185I: The liberty-maven-plugin configuration parameter "appsDirectory" value defaults to "dropins".
[INFO] Installing assembly...
[INFO] Expanding: /Users/liam/.m2/repository/io/openliberty/openliberty-kernel/19.0.0.11/openliberty-kernel-19.0.0.11.zip into /Users/liam/Projects/temp/test1/finish/target/liberty
[INFO] CWWKM2143I: Server defaultServer does not exist. Now creating...
[INFO] CWWKM2001I: Invoke command is [/Users/liam/Projects/temp/test1/finish/target/liberty/wlp/bin/server, create, defaultServer].
[INFO]
[INFO] Server defaultServer created.
[INFO] CWWKM2129I: Server defaultServer has been created at /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer.
[INFO] Copying 1 file to /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer
[INFO] CWWKM2144I: Update server configuration file server.xml from /Users/liam/Projects/temp/test1/finish/src/main/liberty/config/server.xml.
[INFO] Running liberty:install-feature
[INFO] CWWKM2102I: Using artifact based assembly archive : io.openliberty:openliberty-kernel:null:19.0.0.11:zip.
[INFO] CWWKM2102I: Using installDirectory : /Users/liam/Projects/temp/test1/finish/target/liberty/wlp.
[INFO] CWWKM2102I: Using serverName : defaultServer.
[INFO] CWWKM2102I: Using serverDirectory : /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer.
[INFO] Installing features: [jsonp-1.1, jaxrs-2.1]
[INFO] Product validation completed successfully.
[INFO] The following features have been installed: jsonp-1.1 servlet-4.0 jaxrsClient-2.1 jaxrs-2.1
[INFO] Running liberty:deploy
[INFO] CWWKM2102I: Using artifact based assembly archive : io.openliberty:openliberty-kernel:null:19.0.0.11:zip.
[INFO] CWWKM2102I: Using installDirectory : /Users/liam/Projects/temp/test1/finish/target/liberty/wlp.
[INFO] CWWKM2102I: Using serverName : defaultServer.
[INFO] CWWKM2102I: Using serverDirectory : /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer.
[INFO] Copying 1 file to /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer
[INFO] CWWKM2144I: Update server configuration file server.xml from /Users/liam/Projects/temp/test1/finish/src/main/liberty/config/server.xml.
[INFO] CWWKM2185I: The liberty-maven-plugin configuration parameter "appsDirectory" value defaults to "apps".
[INFO] CWWKM2160I: Installing application guide-rest-client-angular.war.xml.
[INFO] Copying 1 file to /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer
[INFO] CWWKM2144I: Update server configuration file server.xml from /Users/liam/Projects/temp/test1/finish/src/main/liberty/config/server.xml.
[INFO] CWWKM2010I: Searching for CWWKF0011I: in /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer/logs/messages.log. This search will timeout after 30 seconds.
[INFO] CWWKM2013I: The file /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer/logs/messages.log being validated does not exist.
[INFO] CWWKM2001I: Invoke command is [/Users/liam/Projects/temp/test1/finish/target/liberty/wlp/bin/server, debug, defaultServer].
[INFO]
[INFO] Listening for transport dt_socket at address: 7777
[INFO] CWWKM2013I: The file /Users/liam/Projects/temp/test1/finish/target/liberty/wlp/usr/servers/defaultServer/logs/messages.log being validated does not exist.
[INFO] Launching defaultServer (Open Liberty 19.0.0.11/wlp-1.0.34.cl191120191031-0300) on Eclipse OpenJ9 VM, version 12.0.1+12 (en_US)
[INFO] [AUDIT   ] CWWKE0001I: The server defaultServer has been launched.
[INFO] [AUDIT   ] CWWKZ0058I: Monitoring dropins for applications.
[INFO] [AUDIT   ] CWWKT0016I: Web application available (default_host): http://lwestby-ws.rchland.ibm.com:9080/
[INFO] [AUDIT   ] CWWKZ0001I: Application guide-rest-client-angular started in 0.335 seconds.
[INFO] [AUDIT   ] CWWKF0012I: The server installed the following features: [jaxrs-2.1, jaxrsClient-2.1, jsonp-1.1, servlet-4.0].
[INFO] [AUDIT   ] CWWKF0011I: The defaultServer server is ready to run a smarter planet. The defaultServer server started in 4.803 seconds.
[INFO] CWWKM2015I: Match number: 1 is [12/4/19, 14:57:58:801 CST] 00000032 com.ibm.ws.kernel.feature.internal.FeatureManager            A CWWKF0011I: The defaultServer server is ready to run a smarter planet. The defaultServer server started in 4.803 seconds..
[INFO] Press the Enter key to run tests on demand.
```

**Comment by evelinec** [comment]
@realModusOperandi Thank you for the finding the problem. The guide is adjusted to fix this issue in #45. Can you review the changes? Thanks.

**Comment by evelinec** [comment]
Fix is in #45
